fusesoc/coremanager.py

```diff
@@ -249,11 +255,12 @@ def eq_vln(this, that):
                 ]
         # Print a warning for all virtual selections that has no concrete requirement selection
         for virtual in virtual_selection.values():
-            logger.warning(
-                "Non-deterministic selection of virtual core {} selected {}".format(
-                    virtual[1], virtual[0]
+            if virtual[0] not in requested_cores:
```

Comment:
This matching process appears to be too strict. A request for a VLNV without the version specified doesn't match against a VLNV with the version included.